oracle_fdw.c

```diff
@@ -732,6 +734,50 @@ _PG_init(void)
 	on_proc_exit(&exitHook, PointerGetDatum(NULL));
 }
 
+/*
+ * Deparse LIMIT clause translated for ORACLE into FETCH FIRST N ROWS ONLY
+ * If OFFSET is set then current offset value is added to the LIMIT value
+ * so that it will give the Oracle optimizer the right clue.
+ */
+char *
+deparseLimit(PlannerInfo *root, struct OracleFdwState *fdwState, RelOptInfo *baserel)
+{
+	StringInfoData limit_clause;
+
+	initStringInfo(&limit_clause);
+
+	if (root->parse->limitCount != NULL && !root->parse->groupClause)
+	{
+		if (IsA(root->parse->limitCount, Const) && !((Const *) root->parse->limitCount)->constisnull)
+		{
+			char *limit_val;
+			char *offset_val;
+
+			appendStringInfoString(&limit_clause, " FETCH FIRST ");
+			limit_val = deparseExpr(
```

Comment:
I am not sure if a call to `deparseExpr` is necessary here.
If it is a `Const` that is not NULL, a simple call to `datumToString` should be enough.
Such a constant has to be a `bigint` anyway.

Reply:
It is necessary, it can be an expression like `LIMIT nitem OFFSET prev_pos;`

Reply:
Hum no I'm wrong the expression result in a constant. However deparseExpr is doing the right job.

Reply:
Yes, it is doing the right job. I just wondered if the job could be done cheaper, but I admit it is not very important.